### PR TITLE
Adding release webhook configuration to global configuration

### PR DESCRIPTION
**Pull request by kasubram**

### CODE REVIEW

File: tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java [hunk 1]
```diff
@@ -225,6 +227,16 @@ public void doConnect(
         dispatch(request, response, body);
     }
 
+    @RequirePOST
+    public void doRmwebhook(
+            final StaplerRequest request,
+            final StaplerResponse response,
+            @StringBodyParameter @Nonnull final String body) {
+        // Send telemetry
+        TelemetryHelper.sendEvent("team-events-rmwebhook", new TelemetryHelper.PropertyMapBuilder().build());
```
Comment (GitHubSriramB) [inline tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java]: where is this telemetry captured? what details get logged in the telemetry?

File: tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java [hunk 2]
```diff
@@ -62,6 +63,7 @@
         eventMap.put("gitPullRequestMerged", new GitPullRequestMergedEvent.Factory());
         eventMap.put("gitPush", new GitPushEvent.Factory());
         eventMap.put("connect", new ConnectHookEvent.Factory());
+        eventMap.put("rmwebhook", new ConnectReleaseWebHookEvent.Factory());
```
Comment (GitHubSriramB) [inline tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java]: rmWebhook?

File: tfs/src/main/java/hudson/plugins/tfs/TeamPluginGlobalConfig.java [hunk 3]
```diff
@@ -108,6 +118,9 @@ public void setUserAccountMapper(UserAccountMapper userAccountMapper) {
     public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
```
Comment (GitHubSriramB) [inline tfs/src/main/java/hudson/plugins/tfs/TeamPluginGlobalConfig.java]: is this called when the project is loaded in UI?

File: tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java [hunk 4]
```diff
@@ -225,6 +227,16 @@ public void doConnect(
         dispatch(request, response, body);
     }
 
+    @RequirePOST
+    public void doRmwebhook(
```
Comment (GitHubSriramB) [inline tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java]: is this called for every REST API call with rmwebhook in the resource? is the name indicative of the registration?

File: tfs/src/main/java/hudson/plugins/tfs/TeamPluginGlobalConfig.java (hunk 3 above)
Comment (kasubram) [reply tfs/src/main/java/hudson/plugins/tfs/TeamPluginGlobalConfig.java]: When the global setting page is loaded

File: tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java (hunk 4 above)
Comment (kasubram) [reply tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java]: This is root for all rmwebhook related changes.. we have register/link not sure what generic name to use. Just using RmWebHook looks good to me as its the root.

File: tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java (hunk 2 above)
Comment (kasubram) [reply tfs/src/main/java/hudson/plugins/tfs/TeamEventsEndpoint.java]: Fixed.

**Pull request merged**